app/models/experimentation/notifications_experiment.rb

```diff
@@ -81,17 +84,24 @@ def monitor
       end
     end
 
+    def messages_report_key
+      if start_time > NEW_MESSAGE_REPORTING_DATE
+        NEW_MESSAGE_REPORTING_KEY
+      else
+        OLD_MESSAGE_REPORTING_KEY
+      end
+    end
+
```

Comment:
Can we do a data migration to get rid of this? Doesn't have to be a part of this PR, can come later. We'd have to rewrite the JSON blobs and key them by template ids instead.

Reply:
We could but there's ~3M membership records so we'd have to think about a way to do it that's sufficiently performant, hence punting it.